app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -58,7 +58,19 @@ XrResult OpenXRInputSource::Initialize()
 
     // Filter mappings
     for (auto& mapping: OpenXRInputMappings) {
-      if (mapping.systemFilter && strcmp(mapping.systemFilter, mSystemProperties.systemName) != 0) {
+      const char* systemFilter = mapping.systemFilter;
+#if defined(PICOXR)
+      // Pico version 5.4.0 changed the system id.
+      if (mapping.controllerType == device::PicoXR) {
+          char buildId[128] = {0};
+          __system_property_get("ro.build.id", buildId);
+          if (!CompareSemanticVersionStrings(buildId, "5.4.0")) {
+              // System version is >= 5.4.0
+              systemFilter = "PICO 4";
+          }
+      }
+#endif
```

Comment:
Hmm this is not future proof. Note that it'll return "PICO 4" for any device from Pico running 5.4. It might be the Pico4 but also the Pico4E or any other they might ship in the future. I think it'd be much safer to do it in reverse, i.e.
1. replace "Pico HMD" by "PICO 4" in the mappings file
2. if it's <5.4.0 then replace system filter by "Pico HMD". That assumes that any device running <5.4 is a Pico4 which is safer.

Reply:
Done.